common/src/test/java/com/microsoft/identity/common/ropc/PasswordGrantTest.java

```diff
@@ -0,0 +1,165 @@
+import com.microsoft.identity.common.internal.util.StringUtil;
+import com.microsoft.identity.internal.testutils.labutils.Credential;
+import com.microsoft.identity.internal.testutils.labutils.Scenario;
+import com.microsoft.identity.internal.testutils.labutils.TestConfigurationQuery;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+@RunWith(RobolectricTestRunner.class)
+public final class PasswordGrantTest {
+
+    private static final String[] SCOPES = {"user.read", "openid", "offline_access", "profile"};
+    private static final String CLIENT_ID = "4b0db8c2-9f26-4417-8bde-3f0e3656f8e0";
+
+    private Scenario getDefaultTestScenario() {
+        final TestConfigurationQuery query = new TestConfigurationQuery();
+        query.userType = "Member";
+        query.isFederated = false;
+        query.federationProvider = "ADFSv4";
+
+        Scenario scenario = Scenario.GetScenario(query);
+        return scenario;
+    }
+
+    private boolean isEmpty(String[] arr) {
+        return arr == null || arr.length == 0;
+    }
+
+    private String convertScopesArrayToString(final String[] scopes) {
+        if (isEmpty(scopes)) {
+            return null;
+        }
+        final Set<String> scopesInSet = new HashSet<>(Arrays.asList(scopes));
+        return StringUtil.convertSetToString(scopesInSet, " ");
+    }
+
+    private TokenRequest createTokenRequest(String[] scopes, String username, String password) {
+        String scope = convertScopesArrayToString(scopes);
+
+        final TokenRequest tokenRequest = new MicrosoftStsTokenRequest();
+        tokenRequest.setClientId(CLIENT_ID);
+        tokenRequest.setScope(scope);
+        tokenRequest.setUsername(username);
+        tokenRequest.setPassword(password);
+        tokenRequest.setGrantType(TokenRequest.GrantTypes.PASSWORD);
+
+        return tokenRequest;
+    }
+
+    @Test
+    public void canPerformROPC() throws IOException {
+        final Scenario scenario = getDefaultTestScenario();
+        final Credential credential = scenario.getCredential();
+
+        final AADTestAuthority aadTestAuthority = new AADTestAuthority();
+        final OAuth2Strategy testStrategy = aadTestAuthority.createOAuth2Strategy();
+
+        final TokenRequest tokenRequest = createTokenRequest(SCOPES, credential.userName, credential.password);
+
+        try {
+            final TokenResult tokenResult = testStrategy.requestToken(tokenRequest);
+
+            assertEquals(true, tokenResult.getSuccess());
+        } catch (ClientException exception) {
+            fail("Unexpected exception.");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void requestFailsIfUsernameNotProvided() throws IOException {
```

Comment:
same here, more meaningful names like `testRopcFailureManagedUserNoUsername and also a comment on top with brief description would be good.

Consider renaming all the test cases to be in similar pattern for readability

Reply:
Why do you want to prefix all tests with the word "test"? The `@test` annotation already indicates that it is a test. 

How about something like this: `ropcFailsForManagedUserIfUsernameNotProvided`
What do you say?

Reply:
mentioned regarding this in another comment